modules/icgc-argo-workflows/payload/alignment/resources/usr/bin/main.py

```diff
@@ -0,0 +1,220 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+ Copyright (c) 2024, Ontario Institute for Cancer Research (OICR).
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published
+ by the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+ Author: Guanqiao Feng <gfeng@oicr.on.ca>
+ """
+
+from datetime import date
+import argparse
+import copy
+import hashlib
+import json
+import os
+import sys
+import uuid
+import yaml
+
+
+def calculate_size(file_path):
+    return os.stat(file_path).st_size
+
+
+def calculate_md5(file_path):
+    md5 = hashlib.md5()
+    with open(file_path, 'rb') as f:
+        for chunk in iter(lambda: f.read(1024 * 1024), b''):
+            md5.update(chunk)
+    return md5.hexdigest()
+
+
+def rename_file(f, payload, rg_count, sample_info, date_str):
+    experimental_strategy = payload['experiment']['experimental_strategy'].lower()
+
+    if f.endswith('.bam'):
+        file_ext = 'bam'
+    elif f.endswith('.bam.bai'):
+        file_ext = 'bam.bai'
+    elif f.endswith('.cram'):
+        file_ext = 'cram'
+    elif f.endswith('.cram.crai'):
+        file_ext = 'cram.crai'
+    else:
+        sys.exit('Error: unknown aligned seq extention: %s' % f)
+
+    aln_type = ''
+    if 'transcriptAlign' in f:
+        aln_type = 'transcriptAlign'
+    else:
+        aln_type = 'genomeAlign'
+
+    new_name = "%s.%s.%s.%s.%s.%s.%s.%s" % (
+        payload['studyId'],
+        sample_info[0]['donor']['donorId'],
+        sample_info[0]['sampleId'],
+        experimental_strategy,
+        aln_type,
+        date_str,
+        'aln',
+        file_ext
+    )
+
```

Comment:
Can we try to make the filename pattern be back compatible with the existing DNA/RNA alignment filename convention?
Refer: https://wiki.oicr.on.ca/display/icgcargotech/File+naming+conventions
For example:
- DNA alignment: <program_id>.<donor_id>.<sample_id>.<wgs|wxs>.<date>.aln.cram
- RNA alignment: <program_id>.<donor_id>.<sample_id>.<rna-seq>.<date>.<aligner>.<data_subtype>.cram

Reply:
Couldn't open the wiki link above. 

updated to match this naming style, does it look good to you @lindaxiang?
TCRB-CA.DO262498.SA622799.rna-seq.20240816.genomeAln.cram
TCRB-CA.DO262498.SA622799.rna-seq.20240816.transcriptAln.cram
TCRB-CA.DO262498.SA622799.dna-seq.20240816.aln.cram

Reply:
updated accordingly